src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/EntitySchemaProcessor.cs

```diff
@@ -151,12 +173,48 @@ public void ProcessChanges()
                     EntityId = lookup.EntityId,
                     SchemaHash = entry.Key.SchemaHash.ConvertFromHex(),
                     Schema = entry.Value.Schema.SborData.Hex.ConvertFromHex(),
+                    IsDeleted = false,
                 };
 
                 _definitionsToAdd.Add(entryDefinition);
 
                 aggregate.EntryIds.Insert(0, entryDefinition.Id);
             }
+
+            foreach (var deletedSchemaHash in change.DeletedSchemaHashes)
+            {
+                var entryLookup = new SchemaDefinitionEntryDbLookup(lookup.EntityId, deletedSchemaHash.ConvertFromHex());
+
+                if (_schemasToBeDeleted.TryGetValue(entryLookup, out var previousEntry))
+                {
+                    var currentPosition = aggregate.EntryIds.IndexOf(previousEntry.Id);
+
+                    if (currentPosition != -1)
+                    {
+                        aggregate.EntryIds.RemoveAt(currentPosition);
+                    }
+                    else
+                    {
+                        throw new UnreachableException($"Unexpected situation where SchemaEntryDefinition with EntityId:{entryLookup.EntityId}, SchemaHash:{deletedSchemaHash} got deleted but wasn't found in aggregate table.");
+                    }
+                }
+                else
+                {
+                    throw new UnreachableException($"Unexpected situation where SchemaEntryDefinition with EntityId:{entryLookup.EntityId}, SchemaHash:{deletedSchemaHash} got deleted but wasn't found in gateway database.");
+                }
+
+                var entryDefinition = new SchemaEntryDefinition
```

Comment:
Ah damn... `*_definition` tables are not meant to keep track of history. They're meant to hold immutable information. 
1. Our queries that rely on `*_definition` tables do not use lateral joins. 
2. Are we sure it is OK to say "schema got deleted" from the network? Is it deletable entity at all? I guess we can say "schema is no longer used by XYZ" (i.e. aggregate) though? Guess we don't need to store `is_deleted` at all?